src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+        haPairs: ontapConfiguration.haPairs,
+        preferredSubnetId: ontapConfiguration.preferredSubnet?.subnetId,
+        routeTableIds: ontapConfiguration.routeTables?.map(routeTable => routeTable.routeTableId),
+        throughputCapacity: ontapConfiguration.throughputCapacity,
+        throughputCapacityPerHaPair: ontapConfiguration.throughputCapacityPerHaPair,
+        weeklyMaintenanceStartTime: ontapConfiguration.weeklyMaintenanceStartTime?.toTimestamp(),
+      },
+      securityGroupIds: [securityGroup.securityGroupId],
+      storageCapacity: props.storageCapacityGiB,
+    });
+
+    return fileSystem;
+  }
+
+  /**
+   * Validates the props provided for a new FSx for Ontap file system.
+   */
+  private validateProps(props: OntapFileSystemProps) {
+    const ontapConfiguration = props.ontapConfiguration;
+    const deploymentType = ontapConfiguration.deploymentType;
+
+    this.validateHaPairs(deploymentType, ontapConfiguration.haPairs);
+    this.validateAutomaticBackupRetention(ontapConfiguration.automaticBackupRetention);
+    this.validateDailyAutomaticBackupStartTime(
+      ontapConfiguration.automaticBackupRetention,
+      ontapConfiguration.dailyAutomaticBackupStartTime,
+    );
+    this.validateDiskIops(props.storageCapacityGiB, ontapConfiguration.diskIops, ontapConfiguration.haPairs);
+    this.validateEndpointIpAddressRange(deploymentType, ontapConfiguration.endpointIpAddressRange);
+    this.validateFsxAdminPassword(ontapConfiguration.fsxAdminPassword);
+    this.validateSubnets(deploymentType, props.vpcSubnets, ontapConfiguration.preferredSubnet);
+    this.validateRouteTables(deploymentType, ontapConfiguration.routeTables);
+    this.validateThroughputCapacity(
+      deploymentType,
+      ontapConfiguration.throughputCapacity,
+      ontapConfiguration.throughputCapacityPerHaPair,
+      ontapConfiguration.haPairs,
+    );
+    this.validateStorageCapacity(ontapConfiguration.haPairs, props.storageCapacityGiB);
+  }
+
+  private validateAutomaticBackupRetention(automaticBackupRetention?: Duration): void {
+    if (
+      automaticBackupRetention == null ||
+      automaticBackupRetention.isUnresolved() ||
+      automaticBackupRetention.toMilliseconds() === 0
+    ) {
+      return;
+    }
+    if (
+      automaticBackupRetention.toMilliseconds() < Duration.days(1).toMilliseconds() ||
+      automaticBackupRetention.toDays() > 90
+    ) {
+      throw new Error('automaticBackupRetention must be between 1 and 90 days or be equal to 0');
+    }
+  }
+
+  private validateDailyAutomaticBackupStartTime(
+    automaticBackupRetention?: Duration,
+    dailyAutomaticBackupStartTime?: aws_fsx.DailyAutomaticBackupStartTime,
+  ): void {
+    if (!dailyAutomaticBackupStartTime) {
+      return;
+    }
+
+    const automaticBackupDisabled = !automaticBackupRetention || automaticBackupRetention?.toDays() === 0;
+
+    if (dailyAutomaticBackupStartTime && automaticBackupDisabled) {
```

Comment:
The default value of `automaticBackupRetention` is 0 in this construct. This means disable automatic backup.

```ts
automaticBackupRetentionDays: ontapConfiguration.automaticBackupRetention?.toDays() ?? 0,
```
https://github.com/open-constructs/aws-cdk-library/pull/42/files#diff-4411deda71b1bb070dac524d0cda1b33091f25bdc5da4d1c5fc8c86468b66d60R278

Because `dailyAutomaticBackupStartTime` can be set for only automatic backup is enabled, I added this validation.